### PR TITLE
chore: coredns place coredns on control plane.

### LINKED ISSUE #64: CoreDNS does not run on the control plane nodes
We should make sure that CoreDNS runs on the control plane nodes, since we can't control it, we can take an approach similar to this:

https://github.com/vexxhost/ansible-collection-kubernetes/blob/main/roles/cluster_api/tasks/patch.yml#L15-L36

### PR DESCRIPTION
**Pull request by mpiscaer**
Fixes #64

### CODE REVIEW

File: roles/coredns/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: coredns
+        namespace: kube-system
+      spec:
+        template:
+          spec:
+            nodeSelector: "{{ coredns_node_selector }}"
+    wait: true
```
Comment (okozachenko1203) [inline roles/coredns/tasks/main.yml:13]: coredns is installed by kubeadm so i suggest that we can move this patch to `kubernetes` role

Comment (okozachenko1203) [reply roles/coredns/tasks/main.yml:13]: i think control-plane.yml file is reasonable? Also please set `run_once: true` in the task.

**Comment by mpiscaer** [comment]
The ci failes there Coredns is not ready, need to start a cluster manual to see what is wrong. Will do this when I have time for it.

**Comment by mpiscaer** [comment]
After running it manual, I get the error:
`0/1 nodes are available: 1 node(s) had untolerated taint {node.kubernetes.io/not-ready: }. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..`

And the node has status `NotReady`.  I think it only gets ready after the role `cilium`.

What is the best way to solve this?

**Review by okozachenko1203 — APPROVED** [review]
@mpiscaer could you rebase this?

**Comment by mpiscaer** [comment]
@okozachenko1203 Yes you can

**Comment by mpiscaer** [comment]
@okozachenko1203 done

File: roles/kubernetes/tasks/control-plane.yml [hunk 2]
```diff
@@ -54,3 +54,17 @@
...
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: coredns
+        namespace: kube-system
+      spec:
+        template:
+          spec:
+            nodeSelector: "{{ coredns_node_selector }}"
+  run_once: true
```
Comment (okozachenko1203) [inline roles/kubernetes/tasks/control-plane.yml:70]:
```suggestion
- name: Set node selector for CoreDNS components
  run_once: true
  kubernetes.core.k8s:
    definition:
      apiVersion: apps/v1
      kind: Deployment
      metadata:
        name: coredns
        namespace: kube-system
      spec:
        template:
          spec:
            nodeSelector: "{{ coredns_node_selector }}"
```

**Pull request merged**